lib/ts/recipe/recipeModule/types.ts

```diff
@@ -1,20 +1,36 @@
-import { NormalisedAppInfo } from "../../types";
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import { NormalisedAppInfo, PostAPIHookFunction } from "../../types";
 
-export type UserInput<GetRedirectionURLContextType, PreAPIHookContextType, OnHandleEventContextType> = {
-    getRedirectionURL?: (context: GetRedirectionURLContextType) => Promise<string | undefined>;
-    preAPIHook?: (context: PreAPIHookContextType) => Promise<{ url: string; requestInit: RequestInit }>;
-    onHandleEvent?: (context: OnHandleEventContextType) => void;
-};
-
-export type Config<GetRedirectionURLContextType, PreAPIHookContextType, OnHandleEventContextType> = {
+export type RecipeConfig<PreAPIHookContext> = {
     recipeId: string;
     appInfo: NormalisedAppInfo;
-} & UserInput<GetRedirectionURLContextType, PreAPIHookContextType, OnHandleEventContextType>;
+    preAPIHook?: (context: PreAPIHookContext) => Promise<{ url: string; requestInit: RequestInit }>;
+    postAPIHook?: PostAPIHookFunction;
+};
 
-export type NormalisedConfig<GetRedirectionURLContextType, PreAPIHookContextType, OnHandleEventContextType> = {
+export type NormalisedRecipeConfig<PreAPIHookContext> = {
     recipeId: string;
     appInfo: NormalisedAppInfo;
-    getRedirectionURL: (context: GetRedirectionURLContextType) => Promise<string | undefined>;
-    preAPIHook: (context: PreAPIHookContextType) => Promise<{ url: string; requestInit: RequestInit }>;
-    onHandleEvent: (context: OnHandleEventContextType) => void;
+    preAPIHook: (context: PreAPIHookContext) => Promise<{ url: string; requestInit: RequestInit }>;
+    postAPIHook: PostAPIHookFunction;
+};
+
+export type RecipeFunctionOptions = {
+    preAPIHook?: (input: {
+        url: string;
+        requestInit: RequestInit;
+    }) => Promise<{ url: string; requestInit: RequestInit }>;
 };
```

Comment:
Comment here saying whey we don't have `postAPIHook` in a recipe function level.